pkg/metrics/metrics.jsx

```diff
@@ -842,24 +843,57 @@
     }
 }
 
-const SvgGraph = ({ data, resource, have_sat }) => {
+const SvgGraph = ({ minute, data, resource, have_sat }) => {
+    const key = (have_sat ? "sat_" : "use_") + resource;
+    const dataFiltered = data.filter(d => d && typeof d[key] === 'number');
+
+    if (!dataFiltered.length)
+        return null;
+
+    const isOverThreshold = dataFiltered.some(sample => sample[key] >= LOAD_THRESHOLD);
     const dataPoints = key => (
         "0,0 " + // start polygon at (0, 0)
-        data.map((samples, index) => (samples && typeof samples[key] === 'number') ? samples[key].toString() + "," + index.toString() : "").join(" ") +
-        " 0," + (data.length - 1) // close polygon
+        dataFiltered.map((samples, index) => Math.min(LOAD_THRESHOLD, samples[key]).toString() + "," + index.toString()).join(" ") +
+        " 0," + (dataFiltered.length - 1) // close polygon
     );
 
+    // Paint graph red when resource usage goes above LOAD_THRESHOLD
+    let firstRed;
+    let lastRed;
+    if (isOverThreshold) {
+        dataFiltered.forEach((d, index) => {
+            if (d[key] > LOAD_THRESHOLD) {
+                if (firstRed === undefined)
+                    firstRed = index;
+                lastRed = index;
+            }
+        });
+    }
+    const dataPointsRed = key => (
+        Math.min(LOAD_THRESHOLD, dataFiltered[firstRed][key]).toString() + "," + firstRed + " " +
+        dataFiltered.map((samples, index) => samples[key] > LOAD_THRESHOLD ? samples[key].toString() + "," + index.toString() : "").join(" ") + " " +
+        Math.min(LOAD_THRESHOLD, dataFiltered[firstRed][key]).toString() + "," + lastRed // close polygon
+    );
+
+    if (isOverThreshold && minute == 12)
+        console.info(minute, dataPointsRed(key));
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18461-20230308-172459-d7e58a47-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)